### PR TITLE
Split issue/pull view router function as multiple smaller functions

### PR DESCRIPTION
**Pull request by lunny**
This PR splits issue/pull view function into multiple smaller functions. It also removed duplicated branches load.

### CODE REVIEW

File: models/issues/issue.go [hunk 1]
```diff
@@ -302,6 +302,7 @@ func (issue *Issue) loadReactions(ctx context.Context) (err error) {
 			comment.Reactions = append(comment.Reactions, react)
 		}
 	}
+	issue.Reactions = reactions
```
Comment (wxiaoguang) [inline models/issues/issue.go]: Why?

image

File: routers/web/repo/issue_view.go [hunk 2]
```diff
@@ -397,24 +374,133 @@ func ViewIssue(ctx *context.Context) {
...
-		role                 issues_model.RoleDescriptor
-		ok                   bool
-		marked               = make(map[int64]issues_model.RoleDescriptor)
-		comment              *issues_model.Comment
-		participants         = make([]*user_model.User, 1, 10)
-		latestCloseCommentID int64
-	)
+	if prepareIssueViewCommentsAndSidebarParticipants(ctx, issue) {
+		return
+	}
```
Comment (wxiaoguang) [inline routers/web/repo/issue_view.go]: That's really bad naming.

`if preparation succeeds, then return`?

File: models/issues/issue.go (hunk 1 above)
Comment (lunny) [reply models/issues/issue.go]: Mistake. 0ef0c5147c

File: routers/web/repo/issue_view.go (hunk 2 above)
Comment (lunny) [reply routers/web/repo/issue_view.go]: The return value means `written`, not success.

File: models/issues/issue.go (hunk 1 above)
Comment (wxiaoguang) [reply models/issues/issue.go]: Why always so many mistakes ........

File: routers/web/repo/issue_view.go (hunk 2 above)
Comment (wxiaoguang) [reply routers/web/repo/issue_view.go]: Readers do not think so, do not bring more surprises to readers. I believe all people with common sense read  `if do something { return }`  as `if something succeeds, then return`

If you are not able to name it clearly, please just check `if ctx.Written()`

Comment (wxiaoguang) [reply routers/web/repo/issue_view.go]: And there is one more problem, due to the problematic web framework design, if developers forget to write response and only "return true/false" from the "prepare" functions, then end users only sees an 200 empty page.

The only right method is to always check "ctx.Written()", avoid abusing "written" and "return"

**Review by wxiaoguang — APPROVED** [review]

**Review by Zettat123 — APPROVED** [review]

**Pull request merged**